packages/driver/src/multi-domain/domain_fn.ts

```diff
@@ -79,18 +81,27 @@ export const handleDomainFn = (cy: $Cy, specBridgeCommunicator: SpecBridgeDomain
       const name = command.get('name')
       const logId = command.getLastLog()?.get('id')
 
-      specBridgeCommunicator.toPrimary('command:end', { id, name, err, logId })
+      specBridgeCommunicator.toPrimaryCommandEnd({ id, name, err, logId })
     })
 
     try {
       // await the eval func, whether it is a promise or not
       // we should not need to transpile this as our target browsers support async/await
       // see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function for more details
-      await window.eval(fnWrapper)(data)
+      const value = window.eval(fnWrapper)(data)
 
-      specBridgeCommunicator.toPrimary('ran:domain:fn')
+      // If we detect a non promise value returned
+      if (value && cy.queue.length > 0 && !value.then) {
```

Comment:
How does this work with assertions? I think this block actually gets skipped and goes to the `else` since there aren't any commands in queue, which works! I wonder if the comment needs more context, like `If we detect a non promise value with commands in queue`
```js
    cy.switchToDomain('foobar.com', () => {
      expect('foo').to.equal('foo')
    })
```

Reply:
This will yield, `undefined`

Reply:
Pretty sure the value is undefined and there wouldn't be any commands in the queue

Reply:
and i'll update the comment :)